### PR TITLE
Extends

### LINKED ISSUE #590: fuse-box ignoring 'extends' in tsconfig.json
Hi, i just noticed that "when a tsconfig file uses "extends" it gets ignored when using fuse-box, yet works correctly when using tsc directly.

For example, say I have these two files:
../tsconfig.json, with compiler options such as "jsx": "React"
./tsconfig.json, with "extends": "../tsconfig.json"

If I run tsc againt .tsx files it works correctly, since ./tsconfig correctly extends the base one.
Fuse-box fails with 
```
  → Typescript config:  \tsconfig.json
Acorn error: Unexpected token (75:51)
```

but if I remove the extends and move the "jsx": "React" to the local file then it works correctly.

### PR DESCRIPTION
**Pull request by unional**
closes #590 

I'm not familiar with the test system so I don't know how to write a test for it.

I have tested this in my own repo usage.

### CODE REVIEW

File: src/core/TypescriptConfig.ts [hunk 1]
```diff
@@ -110,8 +110,8 @@ export class TypescriptConfig {
 				const configFileRelPath = this.configFile.replace(this.context.appRoot, "");
 				this.context.log.echoInfo(`Typescript config file:  ${configFileRelPath}`);
 				configFileFound = true;
-				const res = ts.readConfigFile(this.configFile, p => fs.readFileSync(p).toString());
-				config = res.config;
+				const res = ts.readConfigFile(this.configFile, ts.sys.readFile);
+				config = { ...res.config, compilerOptions: ts.parseJsonConfigFileContent(res.config, ts.sys, this.context.appRoot).options }
```
Comment (unional) [inline src/core/TypescriptConfig.ts]: One thing I'm not sure is whether should I guard against `!res.config`

File: src/core/TypescriptConfig.ts [hunk 2]
```diff
@@ -110,8 +110,8 @@ export class TypescriptConfig {
 				const configFileRelPath = this.configFile.replace(this.context.appRoot, "");
 				this.context.log.echoInfo(`Typescript config file:  ${configFileRelPath}`);
 				configFileFound = true;
-				const res = ts.readConfigFile(this.configFile, p => fs.readFileSync(p).toString());
-				config = res.config;
+				const res = ts.readConfigFile(this.configFile, ts.sys.readFile);
+				config = { ...res.config, compilerOptions: ts.parseJsonConfigFileContent(res.config, ts.sys, this.context.appRoot).options };
```
Comment (unional) [inline src/core/TypescriptConfig.ts]: One thing I'm not sure is whether should I guard against `!res.config`

File: src/core/TypescriptConfig.ts [hunk 3]
```diff
@@ -110,8 +110,8 @@ export class TypescriptConfig {
 				const configFileRelPath = this.configFile.replace(this.context.appRoot, "");
 				this.context.log.echoInfo(`Typescript config file:  ${configFileRelPath}`);
 				configFileFound = true;
-				const res = ts.readConfigFile(this.configFile, p => fs.readFileSync(p).toString());
-				config = res.config;
+				const res = ts.readConfigFile(this.configFile, ts.sys.readFile);
```
Comment (nchanged) [inline src/core/TypescriptConfig.ts]: I think you should use fuse detections of tsconfig. It should be there.

**Comment by unional** [comment]
I notice the current change is not sufficient. Will make another change in the morning.

File: src/core/TypescriptConfig.ts (hunk 3 above)
Comment (nchanged) [reply src/core/TypescriptConfig.ts]: oh maybe you are right. Gimme some time to verify the fix

File: src/core/TypescriptConfig.ts [hunk 4]
```diff
@@ -110,8 +110,8 @@ export class TypescriptConfig {
...
+				config = { ...res.config, compilerOptions: ts.parseJsonConfigFileContent(res.config, ts.sys, this.context.appRoot).options };
 				if (res.error) {
 					this.context.log.echoError(`Errors in ${configFileRelPath}`);
```
Comment (unional) [inline src/core/TypescriptConfig.ts]: Trying to understand how the code works here.

If there are errors, the current code just echo and continue processing.

Is this desirable? I'll need to load the files recursively, so I need to fail the call if one of the files can't be read.

File: src/core/TypescriptConfig.ts [hunk 5]
```diff
@@ -110,8 +110,8 @@ export class TypescriptConfig {
 				const configFileRelPath = this.configFile.replace(this.context.appRoot, "");
 				this.context.log.echoInfo(`Typescript config file:  ${configFileRelPath}`);
 				configFileFound = true;
```
Comment (unional) [inline src/core/TypescriptConfig.ts]: Another concern here.

When there is error, does this still hold true?

**Comment by unional** [comment]
The latest change works.

**Pull request merged**